sys/src/9/ufs/ffs/ffs_vfsops.c

```diff
@@ -773,13 +790,16 @@ ffs_mountfs (struct vnode *devvp, struct mount *mp, struct thread *td)
 		    fs->fs_bsize <= MAXBSIZE &&
 		    fs->fs_bsize >= sizeof(struct fs))
 			break;
-		brelse(bp);
-		bp = nil;
 	}
+
 	if (sblock_try[i] == -1) {
-		error = EINVAL;		/* XXX needs translation */
+		//error = EINVAL;		/* XXX needs translation */
+		// TODO HARVEY error string?
+		error = -1;
```

Comment:
Agreed, though in these cases I'm reluctant to change too much right now.  Also, I don't want to break the flow of code for error handling, especially for deeper errors that might bubble up.